### ISSUE TITLE
All events being filtered out

### ISSUE THREAD

**Issue by Rory-Powell** [plibither8/posthog-filter-out-plugin#3]
## Description

Hi there, I applied the following config:

```json
[
  {
    "property": "$ip",
    "type": "string",
    "operator": "is_not",
    "value": "<some_ip_address>"
  }
]
```

However all new events are dropped with this configuration. Have I done something wrong?

My goal is to drop events from a particular ip address

**Comment by mathieubellon** [comment]
I am with the same issue, when I activate the plugin all events are dropped, with a simple JSON

```JSON
[
  {
    "property": "email",
    "type": "string",
    "operator": "not_contains",
    "value": "loremipsum.com"
  }
]
```

The relation between the JSON we have to upload and the config field we have to fill in are not clear
With the Json schema do I define a specific cohort to exclude AND do I have to specify the events (like $pageview ?). It is my understanding that if a event match what I have in my filter (user email contains loremipsum.com, it is dropped)

I tried everything but all events are dropped whatsoever.

![image](https://github.com/plibither8/posthog-filter-out-plugin/assets/1445897/9fd5b901-d14a-45d2-9ceb-bb106703b508)

**Comment by mathieubellon** [comment]
What's strange is that I see that the filter is actually filtering something in the processEvent metrics  (great feature)
I have no error reported in the plugin logs (great feature also, it is great to have such visibility on plugin internals)

![image](https://github.com/plibither8/posthog-filter-out-plugin/assets/1445897/e2c3b203-726c-4711-8a2e-c8a878b3fafb)

Also i don't understand the signification of the green/gray bullet here ? Does it indicate the deployment of the plugin to my instance ?
![image](https://github.com/plibither8/posthog-filter-out-plugin/assets/1445897/a95f4c54-204e-455d-8b88-1672cd7c8ba3)

**Comment by mathieubellon** [comment]
Yes, I tried to switch "contains" and "not_contains" in the filter config, a basic filter copied from the documentation and each time the events stopped flowing in. The plugin is not working or there is information missing in the documentation

**Comment by MarconLP** [comment]
Hey @mathieubellon, the filter out plugin is only able to filter based on event properties. So that the default config:
```json
[
   {
      "property":"email",
      "type":"string",
      "operator":"not_contains",
      "value":"loremipsum.com"
   }
]
```
will filter out events like: `posthog.capture('test-event', { email: "test2@loremipsum.com" })`

**Comment by MarconLP** [comment]
>Also i don't understand the signification of the green/gray bullet here ? Does it indicate the deployment of the plugin to my instance ?

That indicates the success rate over the last 24 hours: 
![2023-10-19 at 16 17 26@2x](https://github.com/plibither8/posthog-filter-out-plugin/assets/13001502/7fc08474-b0ca-4463-a7be-e50b394c4bad)

**Comment by MarconLP** [comment]
To explain all of the fields:
1. The config: Events have to match every filter in there.
2. Events to filter out: You can define events that will always be dropped.
3. Keep event if filtered property is undefined: That defines if you keep the event in case you filter for an `email` property but the event does not have an `email` property set.
